pkg/autoops/api/api.go

```diff
@@ -879,19 +951,43 @@ func (s *AutoOpsService) isNoUpdateAutoOpsRuleCommand(req *autoopsproto.UpdateAu
 
 func (s *AutoOpsService) createUpdateAutoOpsRuleCommands(req *autoopsproto.UpdateAutoOpsRuleRequest) []command.Command {
 	commands := make([]command.Command, 0)
+
+	// The current web console uses ChangeAutoOpsRuleOpsTypeCommand to update OpsTypes and enable or disable them.
+	// So ActionType is not set and you need to update ActionType according to OpsType.
+	// ToDo: If ChangeAutoOpsRuleOpsTypeCommand is no longer used in the web console,
+	// the following code is no longer needed and should be removed.
+	actionType := autoopsproto.ActionType_UNKNOWN
 	if req.ChangeAutoOpsRuleOpsTypeCommand != nil {
 		commands = append(commands, req.ChangeAutoOpsRuleOpsTypeCommand)
+		if req.ChangeAutoOpsRuleOpsTypeCommand.OpsType == autoopsproto.OpsType_DISABLE_FEATURE {
+			actionType = autoopsproto.ActionType_DISABLE
+		} else if req.ChangeAutoOpsRuleOpsTypeCommand.OpsType == autoopsproto.OpsType_ENABLE_FEATURE {
+			actionType = autoopsproto.ActionType_DISABLE
+		}
 	}
+
 	for _, c := range req.AddOpsEventRateClauseCommands {
+		if c.OpsEventRateClause.ActionType == autoopsproto.ActionType_UNKNOWN {
+			c.OpsEventRateClause.ActionType = actionType
+		}
 		commands = append(commands, c)
 	}
 	for _, c := range req.ChangeOpsEventRateClauseCommands {
+		if c.OpsEventRateClause.ActionType == autoopsproto.ActionType_UNKNOWN {
+			c.OpsEventRateClause.ActionType = actionType
+		}
 		commands = append(commands, c)
 	}
 	for _, c := range req.AddDatetimeClauseCommands {
-		commands = append(commands, c)
+		if c.DatetimeClause.ActionType == autoopsproto.ActionType_UNKNOWN {
+			c.DatetimeClause.ActionType = actionType
+			commands = append(commands, c)
+		}
```

Comment:
Maybe this?
```suggestion
		if c.DatetimeClause.ActionType == autoopsproto.ActionType_UNKNOWN {
			c.DatetimeClause.ActionType = actionType
		}
		commands = append(commands, c)
```

Reply:
Thank you！
I Fixed https://github.com/bucketeer-io/bucketeer/pull/1095/commits/10702ed311855a7d0c77e32de0dd0bbada19045c .